src/stats.rs

```diff
@@ -0,0 +1,86 @@
+use anyhow::Result;
+use itertools::Itertools;
+use osmpbfreader::{OsmId, OsmObj, Tags};
+use std::collections::{BTreeMap, HashMap};
+use std::io;
+
+use crate::util;
+
+pub fn write(relations: &BTreeMap<OsmId, OsmObj>, mut out: impl io::Write) -> Result<()> {
+    let mut count_relations = 0;
+    let mut count_admin = HashMap::<&str, usize>::new();
+    let mut count_boundaries = HashMap::<&str, usize>::new();
+    let mut count_tags = HashMap::<&str, usize>::new();
+    let mut count_types = HashMap::<&str, usize>::new();
+
+    for obj in relations
+        .values()
+        .filter(|obj| util::filter_all_relations(obj))
+    {
+        count_relations += 1;
+
+        let tags = obj.tags();
+
+        add_count(tags, &mut count_admin, "admin_level");
+        add_count(tags, &mut count_boundaries, "boundary");
+        add_count(tags, &mut count_types, "type");
+
+        for tag in tags
+            .keys()
+            .filter(|tag| !matches!(tag.as_str(), "boundary" | "type"))
+        {
+            *count_tags.entry(tag).or_default() += 1;
+        }
+    }
+
+    writeln!(
+        out,
+        "\
+Stats
+--------------------
+Total number of relations: {count_relations}
+
+Administrative levels (count):
+
+{:}
```

Comment:
Ich musste nachschauen, was ein format string `{:}` überhaupt bedeutet, und es sieht aus wie ein [format string, der eine Breite haben kann, die aber nicht specified ist](https://doc.rust-lang.org/std/fmt/). Das macht das gleiche wie einfach `{}`.

```suggestion
{}
```

Weil hier so viele verschiedene values formatiert werden, würde ich trotzdem wie [ursprünglich vorgeschlafen](https://github.com/open-dms/osmtools/pull/4#discussion_r1383753331) temporaries einführen und [named parameters](https://doc.rust-lang.org/std/fmt/) benutzen. Das macht zur Laufzeit genau die gleichen allocations, ist aber IMO leserlicher.

Reply:
Ah okay, kleiner Irrtum